pkg/handler/enqueue_annotation.go

```diff
@@ -0,0 +1,173 @@
+	// to use. It can be any arbitrary value. However, it would be recommended to fill it with `<Kind>.<group>` from
+	// the primary resource to, from a human-readability standpoint, make it easy to tell which specific resource
+	// is interested in. E.g Pods.core or ReplicaSet.apps.
+	TypeAnnotation = "sigs.k8s.io/primary-resource-type"
+)
+
+// EnqueueRequestForAnnotation enqueues Requests based on the presence of annotations that contain the type and
+// namespaced name of the primary resource. The purpose of this handler is to support cross-scope ownership
+// relationships that are not supported by native owner references.
+//
+// This handler should ALWAYS be paired with a finalizer on the primary resource. While the
+// annotation-based watch handler does not have the same scope restrictions that owner references
+// do, they also do not have the garbage collection guarantees that owner references do. Therefore,
+// if the reconciler of a primary resource creates a child resource across scopes not supported by
+// owner references, it is up to the reconciler to clean up that child resource.
+//
+// **NOTE** You might prefer to use the EnqueueRequestsFromMapFunc with predicates instead of it.
+//
+// **Examples:**
+//
+// The following code will enqueue a Request to the `primary-resource-namespace` when some change occurs in a Pod
+// resource:
+//
+// ...
+// if err := c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForAnnotation{"Pods.core"}); err != nil {
+//    entryLog.Error(err, "unable to watch Pods")
+//    os.Exit(1)
+// }
+// ...
+//
+// With the annotations:
+//
+// ...
+// annotations:
+//    sigs.k8s.io/primary-resource-name: "my-namespace/my-pod"
+//    sigs.k8s.io/primary-resource-type: "Pods.core"
+// ...
+//
+// The following code will enqueue a Request to the `primary-resource-namespace`, ReplicaSet reconcile,
+// when some change occurs in a ClusterRole resource
+//
+// ...
+// if err := c.Watch(&source.Kind{
+//	  // Watch cluster roles
+//	  Type: &rbacv1.ClusterRole{}},
+//
+//	  // Enqueue ReplicaSet reconcile requests using the
+//	  // namespacedName annotation value in the request.
+//	  &handler.EnqueueRequestForAnnotation{"ReplicaSet.apps"}); err != nil {
+//	      entryLog.Error(err, "unable to watch ClusterRole")
+//	      os.Exit(1)
+//    }
+// }
+// ...
+// With the annotations:
+//
+// ...
+// annotations:
+//    sigs.k8s.io/primary-resource-name: "my-namespace/my-replicaset"
+//    sigs.k8s.io/primary-resource-type: "ReplicaSet.apps"
+// ...
+type EnqueueRequestForAnnotation struct {
+	// It is used to verify that the primary resource is the primary resource to use. It can be any arbitrary value.
+	// However, it would be recommended to fill it with `<Kind>.<group>` from the primary resource to make it easy
+	// to tell which specific resource is interested in.
+	Type string
```

Comment:
IMO this'd better to be a `schema.GroupKind` unless there are some use cases that requires it be to an arbitrary string.
It's easier for users to get this right if it's more structured and you also provide a few helper functions.
I imagine that if this is `GroupKind` then we can at least have one helper function to set namespacedName and GroupKind. e.g.`func SetOwnerAnnotation(owner, owned runtime.Object) error`

Reply:
@DirectXMan12 @droot @joelanford @vincepri WDYT?

Reply:
I agree with your suggestion. I am working on to address it.

Reply:
Done @mengqiy let us know wdyt..